commands/update/nodepool/command.go

```diff
@@ -113,10 +132,22 @@ func verifyPreconditions(args Arguments) error {
 		return microerror.Mask(errors.ClusterNameOrIDMissingError)
 	} else if args.NodePoolID == "" {
 		return microerror.Mask(errors.NodePoolIDMissingError)
-	} else if args.ScalingMin == 0 && args.ScalingMax == 0 && args.Name == "" {
-		return microerror.Mask(errors.NoOpError)
-	} else if args.ScalingMin > args.ScalingMax && args.ScalingMax > 0 {
-		return microerror.Mask(errors.WorkersMinMaxInvalidError)
+	}
+
+	if args.ScalingMin == 0 && args.ScalingMax == 0 && args.Name == "" {
+		return microerror.Maskf(errors.NoOpError, "Nothing to update.")
+	}
+
+	switch args.Provider {
+	case provider.AWS:
+		if args.ScalingMin > args.ScalingMax && args.ScalingMax > 0 {
+			return microerror.Mask(errors.WorkersMinMaxInvalidError)
+		}
+
+	case provider.Azure:
+		if args.ScalingMin != args.ScalingMax {
+			return microerror.Maskf(errors.NoOpError, "Provider '%s' does not support node pool autoscaling.", args.Provider)
```

Comment:
Isn't `WorkersMinMaxInvalidError` for validation error?

```suggestion
			return microerror.Maskf(errors.WorkersMinMaxInvalidError, "Provider '%s' does not support node pool autoscaling.", args.Provider)
```

Reply:
Nice one! Thank you 👍